pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,382 @@
+			q := &survey.Question{
+				Name: inputName,
+				Prompt: &survey.Input{
+					Message: inputName,
+					Default: input.Default,
+				},
+			}
+			if input.Required {
+				q.Validate = survey.Required
+			}
+			qs = append(qs, q)
+		}
+
+		sort.Slice(qs, func(i, j int) bool {
+			return qs[i].Name < qs[j].Name
+		})
+
+		inputAnswer := InputAnswer{
+			providedInputs: providedInputs,
+		}
+		err = prompt.SurveyAsk(qs, &inputAnswer)
+		if err != nil {
+			return err
+		}
+	} else {
+		if opts.JSON != "" {
+			err := json.Unmarshal([]byte(opts.JSON), &providedInputs)
+			if err != nil {
+				return fmt.Errorf("could not parse provided JSON: %w", err)
+			}
+		}
+
+		if len(opts.InputArgs) > 0 {
+			fs := pflag.FlagSet{}
+			for inputName, input := range inputs {
+				fs.String(inputName, input.Default, input.Description)
+			}
+			err = fs.Parse(opts.InputArgs)
+			if err != nil {
+				return fmt.Errorf("could not parse input args: %w", err)
+			}
+			for inputName := range inputs {
+				providedValue, _ := fs.GetString(inputName)
+				providedInputs[inputName] = providedValue
+			}
+		}
+	}
+
+	for inputName, inputValue := range inputs {
+		if inputValue.Required && providedInputs[inputName] == "" {
+			return fmt.Errorf("missing required input '%s'", inputName)
+		}
+	}
+
+	path := fmt.Sprintf("repos/%s/actions/workflows/%d/dispatches",
+		ghrepo.FullName(repo), workflow.ID)
+
+	requestByte, err := json.Marshal(struct {
+		Ref    string            `json:"ref"`
+		Inputs map[string]string `json:"inputs"`
+	}{
+		Ref:    ref,
+		Inputs: providedInputs,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serialize workflow inputs: %w", err)
+	}
+
+	body := bytes.NewReader(requestByte)
+
+	err = client.REST(repo.RepoHost(), "POST", path, body, nil)
+	if err != nil {
+		return fmt.Errorf("could not create workflow dispatch event: %w", err)
+	}
+
+	if opts.IO.CanPrompt() {
+		out := opts.IO.Out
+		cs := opts.IO.ColorScheme()
+		fmt.Fprintf(out, "%s Created workflow_dispatch event for %s at %s\n",
+			cs.SuccessIcon(), cs.Cyan(workflow.Base()), cs.Bold(ref))
+
+		fmt.Fprintln(out)
+
+		fmt.Fprintf(out, "To see runs for this workflow, try: %s\n",
+			cs.Boldf("gh run list --workflow=%s", workflow.Base()))
+	}
+
+	return nil
+}
+
+type WorkflowInput struct {
+	Required    bool
+	Default     string
+	Description string
+}
+
+func findInputs(yamlContent []byte) (map[string]WorkflowInput, error) {
+	var rootNode yaml.Node
+	err := yaml.Unmarshal(yamlContent, &rootNode)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse workflow YAML: %w", err)
+	}
+
+	if len(rootNode.Content) != 1 {
+		return nil, errors.New("invalid yaml file")
+	}
+
+	var onKeyNode *yaml.Node
+	var dispatchKeyNode *yaml.Node
+	var inputsKeyNode *yaml.Node
+	var inputsMapNode *yaml.Node
+
+	// TODO this is pretty hideous
+	for _, node := range rootNode.Content[0].Content {
+		if onKeyNode != nil {
+			for _, node := range node.Content {
```

Comment:
This looks like it might fail to parse `on: workflow_file`, which I believe is valid workflow syntax